assignment of read-only location in cpp file compiled with nrnivmodl-core

I got a mod file that gave me this error:

```
x86_64/core/mod2c/SK_E2.cpp:241:30: error: assignment of read-only location '*(inst->coreneuron::SK_E2_Instance::ek + ((sizetype)(((long unsigned int)id) * 8)))'
  241 |                 inst->ek[id] = inst->ion_ek[indexes[0*pnodecount+id]];
      |                 ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x86_64/core/mod2c/SK_E2.cpp:242:31: error: assignment of read-only location '*(inst->coreneuron::SK_E2_Instance::cai + ((sizetype)(((long unsigned int)id) * 8)))'
  242 |                 inst->cai[id] = inst->ion_cai[indexes[3*pnodecount+id]];
      |                 ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x86_64/core/mod2c/SK_E2.cpp: In function 'void coreneuron::nrn_cur_SK_E2(coreneuron::NrnThread*, coreneuron::Memb_list*, int)':
x86_64/core/mod2c/SK_E2.cpp:286:26: error: assignment of read-only location '*(inst->coreneuron::SK_E2_Instance::ek + ((sizetype)(((long unsigned int)id) * 8)))'
  286 |             inst->ek[id] = inst->ion_ek[indexes[0*pnodecount+id]];
      |             ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x86_64/core/mod2c/SK_E2.cpp:287:27: error: assignment of read-only location '*(inst->coreneuron::SK_E2_Instance::cai + ((sizetype)(((long unsigned int)id) * 8)))'
  287 |             inst->cai[id] = inst->ion_cai[indexes[3*pnodecount+id]];
      |             ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x86_64/core/mod2c/SK_E2.cpp: In function 'void coreneuron::nrn_state_SK_E2(coreneuron::NrnThread*, coreneuron::Memb_list*, int)':
x86_64/core/mod2c/SK_E2.cpp:317:26: error: assignment of read-only location '*(inst->coreneuron::SK_E2_Instance::ek + ((sizetype)(((long unsigned int)id) * 8)))'
  317 |             inst->ek[id] = inst->ion_ek[indexes[0*pnodecount+id]];
      |             ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x86_64/core/mod2c/SK_E2.cpp:318:27: error: assignment of read-only location '*(inst->coreneuron::SK_E2_Instance::cai + ((sizetype)(((long unsigned int)id) * 8)))'
  318 |             inst->cai[id] = inst->ion_cai[indexes[3*pnodecount+id]];
      |             ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x86_64/core/mod2c/SK_E2.cpp: In function 'void* coreneuron::mem_alloc(size_t, size_t, size_t)':
x86_64/core/mod2c/SK_E2.cpp:133:23: warning: ignoring return value of 'int posix_memalign(void**, size_t, size_t)', declared with attribute warn_unused_result [-Wunused-result]
  133 |         posix_memalign(&ptr, alignment, num*size);
      |         ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
make: *** [/home/katta/projects/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/coreneuron-develop-2z6mo3/share/coreneuron/nrnivmodl_core_makefile:148: x86_64/core/build/SK_E2.o] Error 1
make: *** Waiting for unfinished jobs....
```
this error is a recent development since the mod file was compiling just fine before. With mod2c there are no problems.

--- a/src/codegen/codegen_c_visitor.cpp
+++ b/src/codegen/codegen_c_visitor.cpp
@@ -731,7 +731,10 @@ bool CodegenCVisitor::is_constant_variable(const std::string& name) const {
     auto symbol = program_symtab->lookup_in_scope(name);
     bool is_constant = false;
     if (symbol != nullptr) {
-        if (symbol->has_any_property(NmodlType::param_assign) && symbol->get_write_count() == 0) {
+        // per mechanism ion variables needs to be updated from neuron/coreneuron values
+        if (info.is_ion_variable(name)) {
+            is_constant = false;
+        } else if (symbol->has_any_property(NmodlType::param_assign) && symbol->get_write_count() == 0) {
             is_constant = true;
         }
     }

--- a/src/codegen/codegen_c_visitor.cpp
+++ b/src/codegen/codegen_c_visitor.cpp
@@ -734,7 +734,8 @@ bool CodegenCVisitor::is_constant_variable(const std::string& name) const {
         // per mechanism ion variables needs to be updated from neuron/coreneuron values
         if (info.is_ion_variable(name)) {
             is_constant = false;
-        } else if (symbol->has_any_property(NmodlType::param_assign) && symbol->get_write_count() == 0) {
+        } else if (symbol->has_any_property(NmodlType::param_assign) &&
+                   symbol->get_write_count() == 0) {
             is_constant = true;
         }
     }

